### PR TITLE
Segment and Pick Anything

### PR DESCRIPTION
**Pull request by chancecardona**
This is pick object for an arbitrary object, as assisted by SuperSeg server. As seen on demo-gifs.

Not sure if we want this in here per se, but this is cool functionality.

### CODE REVIEW

File: src/picknik_ur_site_config/objectives/sam_pick_object.xml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<root BTCPP_format="4" main_tree_to_execute="SAM Pick Object">
+  <!--//////////-->
+  <BehaviorTree ID="SAM Pick Object" _description="Uses SAM to Segment out an arbitrary object and pick it.">
```
Comment (pac48) [inline src/picknik_ur_site_config/objectives/sam_pick_object.xml:4]:
```suggestion
  <BehaviorTree ID="SAM Pick Object" _description="Uses SAM to segment out an arbitrary object and pick it.">
```

File: src/picknik_ur_site_config/objectives/sam_pick_object.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      <Action ID="GetTextFromUser" text_output="{text_output}" prompts="Enter the name of the object you wish to pick." default_values="rubber duck"/>
+      <Action ID="GetMasks2DAction" image="{image}" action_name="get_masks_2d_superseg" min_confidence="0.8" max_nms_iou="0.8" min_relative_area="0.0002" max_relative_area="0.25" timeout_sec="15.0" masks2d="{masks2d}" valid_classes="{text_output}"/>
+      <Action ID="GetMasks3DFromMasks2D" point_cloud="{point_cloud}" masks2d="{masks2d}" camera_info="{camera_info}" masks3d="{masks3d}"/>
+      <Decorator ID="ForEachMask3D" vector_in="{masks3d}" out="{mask3d}">
+        <Control ID="Sequence">
+          <Action ID="GetPointCloudFromMask3D" point_cloud="{point_cloud}" mask3d="{mask3d}" point_cloud_fragment="{point_cloud_fragment}"/>
+          <Action ID="SendPointCloudToUI" point_cloud="{point_cloud_fragment}" sensor_name="scene_scan_camera" pcd_topic="/pcd_pointcloud_captures"/>
+        </Control>
+      </Decorator>
+      <!--Convert 3D mask into graspable object, but as we don't need it for MTC-->
```
Comment (pac48) [inline src/picknik_ur_site_config/objectives/sam_pick_object.xml:24]: I'm a little confused by this comment, maybe "but" should be removed?

Comment (chancecardona) [inline src/picknik_ur_site_config/objectives/sam_pick_object.xml:24]:
```suggestion
      <!--Convert 3D masks into graspable objects-->
```

**Review by dyackzan — COMMENTED** [review]
Could you add an update on the status of this @chancecardona ? Is this something you want to get merged? And is it in a state ready to be merged? I remember seeing the demo GIF where the objective accurately identified an object using SAM but failed to grasp it when it attempted to pick it

**Comment by chancecardona** [comment]
@dyackzan it should definitely be merged. This does show how AI assisted pick and place works. The grasp is another problem that affects all of MoveIt Pro we're going to need to tackle in another PR.

**Comment by dyackzan** [comment]
Is there documentation on how to run this? I think I need to pull an external image for the ml stuff and rebuild but I'm not sure how. And if that doesn't exist could you add it?

**Comment by chancecardona** [comment]
@dyackzan you're going to need to follow the steps here https://github.com/pickNikRobotics/moveit_studio_segment_anything (private) but nothing else here is required. This other stuff is todo in the future.

**Comment by dyackzan** [comment]
Have been wrestling with getting 002 working and I finally got MoveIt Pro launching yesterday with https://github.com/PickNikRobotics/picknik_ur5e_configs/pull/80. Unfortunately I'm still blocked on testing this because 002 is in a "recoverable fault" state I am unable to get it out of

**Review by dyackzan — APPROVED** [review]
Yayy finally got this tested! Thanks for your patience @chancecardona

**Pull request merged**